compiler/rustc_codegen_gcc/src/intrinsic/mod.rs

```diff
@@ -551,7 +551,23 @@ impl<'gcc, 'tcx> ArgAbiExt<'gcc, 'tcx> for ArgAbi<'tcx, Ty<'tcx>> {
                 bx.lifetime_start(llscratch, scratch_size);
 
                 // ... where we first store the value...
-                bx.store(val, llscratch, scratch_align);
+                if let Some(offset_from_start) = cast.rest_offset {
+                    assert!(cast.prefix[1..].iter().all(|p| p.is_none()));
+                    assert_eq!(cast.rest.unit.size, cast.rest.total);
+                    assert!(cast.prefix[0].is_some());
+                    let first = bx.extract_value(val, 0);
+                    let second = bx.extract_value(val, 1);
+                    bx.store(first, llscratch, scratch_align);
+                    let second_ptr =
+                        bx.inbounds_ptradd(llscratch, bx.const_usize(offset_from_start.bytes()));
+                    bx.store(
+                        second,
+                        second_ptr,
+                        scratch_align.restrict_for_offset(offset_from_start),
+                    );
+                } else {
+                    bx.store(val, llscratch, scratch_align);
+                };
```

Comment:
Ngh... is it too much work to factor out this code into cg_ssa? It gets duplicated in cg_llvm but maybe that can't be avoided.